R/method-correlation.R

```diff
@@ -59,9 +59,205 @@ setMethod("correlation", signature("data.table", "data.table"), function(data1,
+
+  veupathUtils::logWithTime(paste("Received df table with", nrow(abundances), "samples and", (ncol(abundances)-1), "features with abundances."), verbose)
+
+  result <- buildCorrelationComputeResult(corrResult, data1, data1@sampleMetadata, method, verbose)
+  return(result)  
+})
+
+#' Self Correlation
+#'
+#' This function returns correlation coefficients for variables in one dataset against itself
+#' 
+#' @param data1 first dataset. An AbundanceData object or data.table
+#' @param method string defining the type of correlation to run. The currently supported values are 'spearman' and 'pearson'
+#' @param verbose boolean indicating if timed logging is desired
+#' @return ComputeResult object
+#' @import veupathUtils
+#' @export
+setGeneric("selfCorrelation",
+  function(data1, method = c('spearman','pearson'), verbose = c(TRUE, FALSE), ...) standardGeneric("selfCorrelation"),
+  signature = c("data1")
+)
+
+#' Self Correlation
+#'
+#' This function returns correlation coefficients for variables in one AbundanceData object against itself
+#' 
+#' @param data1 An AbundanceData object
+#' @param method string defining the type of correlation to run. The currently supported values are 'spearman' and 'pearson'
+#' @param verbose boolean indicating if timed logging is desired
+#' @return ComputeResult object
+#' @import veupathUtils
+#' @export
+setMethod("selfCorrelation", signature("AbundanceData"), function(data1, method = c('spearman','pearson'), verbose = c(TRUE, FALSE)) {
+  abundances <- getAbundances(data1, FALSE, FALSE)
+  corrResult <- correlation(abundances, method=method, verbose=verbose)
+
+  veupathUtils::logWithTime(paste("Received df table with", nrow(abundances), "samples and", (ncol(abundances)-1), "features with abundances."), verbose)
+
+  result <- buildCorrelationComputeResult(corrResult, data1, NULL, method, verbose)
+  return(result)  
+})
+
+#' Self Correlation
+#'
+#' This function returns correlation coefficients for variables in one SampleMetadata object against itself
+#' 
+#' @param data1 SampleMetadata object
+#' @param method string defining the type of correlation to run. The currently supported values are 'spearman' and 'pearson'
+#' @param verbose boolean indicating if timed logging is desired
+#' @return ComputeResult object
+#' @import veupathUtils
+#' @export
+setMethod("selfCorrelation", signature("SampleMetadata"), function(data1, method = c('spearman','pearson'), verbose = c(TRUE, FALSE)) {
+  corrResult <- correlation(getSampleMetadata(data1, TRUE, FALSE), method=method, verbose=verbose)
+
+  veupathUtils::logWithTime(paste("Received df table with", nrow(data1), "samples and", (ncol(data1)-1), "variables."), verbose)
+
+  result <- buildCorrelationComputeResult(corrResult, data1, NULL, method, verbose)
+  return(result)
+})
+
+#' Self Correlation
+#'
+#' This function returns correlation coefficients for variables in one data.table against itself.
```

Comment:
how does this function differ from the correlation function above with signature `c('data.table', missing)`?

Reply:
Like why have a also the correlation function with that signature, when in these selfCorrelation functions we could just say `correlation(data1, data1)`?

Reply:
no scratch the above i had another idea.

Reply:
it doesnt. its just for convenience. i almost didnt include it, but then i thought itd be a reasonable thing for someone to try to do/ assume existed so i added it.

Reply:
whats the other idea?

Reply:
the other idea was putting the self-correlation-specific deduplication of correlation results in the correlation method with signature data.table, missing